recipes/cmake/3.x.x/test_package/conanfile.py

```diff
@@ -14,10 +14,14 @@ def requirements(self):
     def test(self):
         if can_run(self):
             output = StringIO()
-            self.run("cmake --version", env="conanrun", output=output)
+            # Third arg to self.run renamed "stdout" in Conan 2.0 but 1.x linter doesn't like it
+            self.run("cmake --version", output, env="conanrun")
             output_str = str(output.getvalue())
             self.output.info("Installed version: {}".format(output_str))
-            require_version = str(self.deps_cpp_info["cmake"].version)
+            if Version(conan_version).major < 2:
```

Comment:
Hi @uilianries ,

I agree that having separate test files is preferable and, in fact, that what I thought we had, but CCI appears to want to lint and run the v2 test_package (a.k.a. test_package) using Conan 1.x. How does one tell CCI to only apply Conan 2 logic (linting and execution) to the v2 test and not force it to also comply with Conan v1 without conditional logic? From what I can tell, self.deps_cpp_info simply does not exist in Conan 2.0 and conversely, self.dependencies does not exist in Conan 1.x. Thanks

Reply:
Use self.dependencies on test_package and self.deps_cpp_info test_v1_package

Reply:
Hi @uilianries,

That's exactly what my original submission had (please see this diff) but the CCI bot failed the build because it is using Conan 1.52 on test_package and Conan 1.52 does not support self.dependencies. I fear that if I do what you recommend, I will be back to failing everywhere. It has taken over two days for the current set of changes to be built out across all of the CCI platforms.

Reply:
@System-Arch Indeed `self.dependencies` can not be used by `test()` it's only available on `validate()` and `generate()`: https://docs.conan.io/en/latest/reference/conanfile/dependencies.html#dependencies-interface

It's not 1.52 limitation, but a design concept.

Thus, keep `self.deps_cpp_info`. I'll open an issue trying to update it, your case is relevant as more tools can use the same idea.

Reply:
Hi @uilianries,

Regarding #1241, which now references #11940, it appears to confirm that, although undocumented, self,dependencies does work in test() with Conan 2.0, which matches my experience and explains how I arrived at its use as my initial approach. The history of these issues seems to indicate that there are no plans to add support for using self.dependencies in test() with Conan 1.x, thus I think the only option here is to use conditional code or to eliminate this check from the test_package recipe. Do you agree? Thanks,

Reply:
No, again, we are not using conditions. If it's incompatible, please, open an issue to https://github.com/conan-io/conan/issues asking for backward compatibility. Like I said before, there are tools which are broken for Conan 2.0 but documented for migration. For your specific case, just avoid checking the output, we rarely do it actually. You can execute `cmake --help` and it should be enough. The is testing the package integrity, and in case something goes wrong, it should raise a runtime error.

Reply:
I found I could parse self.tested_reference_str to obtain the expected CMake version number in a Conan version agnostic manner